docs/src/main/tut/datatypes/freeapplicative.md

```diff
@@ -164,5 +164,24 @@ val prodCompiler: FunctionK[ValidationOp, ValidateAndLog] = parCompiler and logC
 val prodValidation = prog.foldMap[ValidateAndLog](prodCompiler)
 ```
 
+### The way FreeApplicative#foldMap works
+Despite being an imperative loop, there is a functional intuition behind `FreeApplicative#foldMap`.
+
+The new `FreeAp`'s `foldMap` is a sort of mutually-recursive function that operates on an argument stack and a 
```

Comment:
quick nitpick question: shall we use `FreeAp` or `FreeApplicative` in the documentation?